src/modules/admin/components/forms/FormTreeView.tsx

```diff
@@ -0,0 +1,267 @@
+import { SvgIconComponent } from '@mui/icons-material';
+import { Badge, lighten, Typography } from '@mui/material';
+import { Box, Stack } from '@mui/system';
+import {
+  TreeItem2Content,
+  TreeItem2GroupTransition,
+  TreeItem2Icon,
+  TreeItem2IconContainer,
+  TreeItem2Label,
+  TreeItem2Provider,
+  TreeItem2Root,
+  UseTreeItem2Parameters,
+} from '@mui/x-tree-view';
+import { RichTreeView } from '@mui/x-tree-view/RichTreeView';
+import { useTreeItem2 } from '@mui/x-tree-view/useTreeItem2/useTreeItem2';
+import React, { useMemo } from 'react';
+import Loading from '@/components/elements/Loading';
+import {
+  FormBooleanIcon,
+  FormChoiceIcon,
+  FormCurrencyIcon,
+  FormDateIcon,
+  FormDisplayIcon,
+  FormFileIcon,
+  FormGroupIcon,
+  FormImageIcon,
+  FormIntegerIcon,
+  FormObjectIcon,
+  FormStringIcon,
+  FormTextIcon,
+  FormTimeOfDayIcon,
+} from '@/components/elements/SemanticIcons';
+import theme from '@/config/theme';
+import { FormDefinitionJson, ItemType } from '@/types/gqlTypes';
+
```

Comment:
I'm sure there will end up being lots of components associated with the form builder. I think it would be good to have its own directory for form builder components.

Either
* `src/modules/admin/components/forms/formBuilder/<components>`
or
* `src/modules/formBuilder/components/<components>`

I think I prefer adding an entirely new module just for the form builder, since there will probably be util functions and maybe hooks associated with it. In my eyes it is very distinct from the rest of the application, so encapsulating that inside a module seems right.